backend/middlewares/validators/lessonValidators.ts

```diff
@@ -27,9 +28,44 @@ export const lessonRequestDtoValidator = async (
   if (!validatePrimitive(body.image, "string")) {
     return res.status(400).send(getApiValidationError("image", "string"));
   }
-  // TO DO: Write better validator for content
+  // content validation
+  if(!validateContentList(body.content)){
+    return res.status(400).send("There was an error in the lesson content.")
+  }
   if (typeof body.content !== "object") {
     return res.status(400).send("There was an error");
   }
   return next();
 };
+
+
+export const validateContent = (content: ContentBlock): boolean => {
+  if (content.type === undefined || content.type === null) return false;
+
+  switch (content.type) {
+    case "text": {
+      let text = content as TextBlock;
+      return text.content.text != null;
+    }
+    case "image": {
+      let image = content as ImageBlock;
+      return image.content.link != null;
+    }
+    case "video": {
+      let video = content as VideoBlock;
+      return video.content.link != null;
+    }
+    default: {
+      return false;
+    }
+  }
+};
+
+export const validateContentList = (contentList: ContentBlock[]): boolean => {
+  contentList.forEach((contentBlock) => {
+    if(!validateContent(contentBlock)){
+      return false;
+    }
+  });
+  return true;
+}
```

Comment:
```suggestion
export const validateContentList = (contentList: ContentBlock[]): boolean =>
  contentList.some((contentBlock) => !validateContent(contentBlock));
```

(may want to double check that logic)